### DISCUSSION TITLE
Is it possible to use a public/private azure devops hosted git datasource using git protocol (not http/https)?

### DISCUSSION THREAD

**Discussion by cschar** [netbox-community/netbox#13830] [Q&A]
I have a public azure devops repository hosted on dev.azure.com

I would like to sync it as a datasource , and am wondering, instead of using http/https , can I use git as the data-source type?
https://dev.azure.com/csc321/OpenSourceProject1/_git/my-sample-datasource?path=/prefixes_1.csv is an example file

```
prefix,status,Children,VRF,Utilization,Tenant,Site,VLAN,Role,Description,ID,Prefix (Flat),Tenant Group,VLAN Group,Pool,Marked Utilized,Comments,Tags,Created,Last updated,Depth
198.168.11.0/24,active,0,Global,0.0%,,,,,,12,198.168.11.0/24,,,False,False,,,2023-09-19 23:11,2023-09-19 23:11,0
198.168.12.0/24,active,0,Global,0.0%,,,,,,13,198.168.12.0/24,,,False,False,,,2023-09-19 23:11,2023-09-19 23:11,0
```

here is my config 
<img width="809" alt="Screenshot 2023-09-19 at 9 20 32 PM" src="https://github.com/netbox-community/netbox/assets/296551/1df026b4-a4d2-426b-8ab8-b7de919738dd">

In the example docs it only shows an https item: https://docs.netbox.dev/en/stable/models/core/datasource/

when i try to sync, it errors, and in the rq service i get 
`sudo journalctl --unit netbox-rq.service -n 100 --no-pager`

```
python3[282812]: 01:22:21 default: core.jobs.sync_datasource(job=<Job: dcacf9f4-3b49-4636-af54-401b7b810f9f>) (dcacf9f4-3b49-4636-af54-401b7b810f9f)
 python3[285618]: ERROR:root:Fetching remote data failed (HangupException): Host key verification failed.
```

How can I configure this ssh datasource to use the public git repo?

### Followup Question:   
- Can I also sync to a private project without relying on HTTP/HTTPS Credentials? Can I continue using SSH somehow?
- Will netbox support Azure Storage accounts in future as a datsource?

**Comment by aascedu** [comment]
UP. I would Like to connect to a git with ssh. Anyone have an idea ?
Thanks

**Comment by candlerb** [comment]
Others have got this working: #14941, https://github.com/netbox-community/netbox/issues/15394#issuecomment-2016409907, https://github.com/netbox-community/netbox/issues/15394#issuecomment-2125648767

Based on your error message, it seems to have gotten as far as opening the ssh connection but doesn't recognize the host key presented by the remote side. Adding the host key to `~netbox/.ssh/known_hosts` might be sufficient.

**Reply by aascedu** [reply]
Is there a .ssh file inside of netbox installation ?
Sorry if the question might be trivial, I am new to the whole application and I am using Docker to deploy it so it makes things confusing from time to time.
Thank you !

**Reply by candlerb** [reply]
I was talking about the specific problem that @cschar was having. But now I see that's a very old question - it's annoying to have an ancient thread resurrected like that.

Your problem seems to be different: you've not even started. Docker complicates things. I can't give you a tested answer for that case.

If you end up with the same "Host key verification failed" error then I'm guessing that first you'll have to see what user Netbox is running as in the container (is there a 'netbox' user, or is it running as 'root'?) Then you'll have to find the home directory of that user, in `/etc/passwd`. Then you'll have to create a `.ssh` subdirectory under the home directory, and add the `known_hosts` file there, with the correct permissions. Which, being docker, is probably best done via a volume mount.

But if you're having some other problem, then obviously nobody can help you without seeing your configuration and error message(s). And in that case, it would be better for you to open a new thread giving all the details.

**Comment by aascedu** [comment]
Thank you for the roadmap to try something, I can understand « resurrecting
» an old thread like that can be annoying.

I will open a thread if I still struggle.

Thank you,
Arthur
